src/main/java/gov/nasa/pds/tools/validate/rule/pds4/TableDataContentValidationRule.java

```diff
@@ -426,23 +431,55 @@ public void validateTableDataContents() throws MalformedURLException,
           } else {  // We have either a character or delimited table
             boolean manuallyParseRecord = false;
             String line = reader.readNextLine();
+            int lineNumber = 0;
             while (line != null) {
               progressCounter();
-              
-              if (!line.endsWith("\r\n")) {
-                addTableProblem(ExceptionType.ERROR,
-                    ProblemType.MISSING_CRLF,
-                    "Record does not end in carriage-return line feed.", 
-                    dataFile, tableIndex, reader.getCurrentRow());
-                manuallyParseRecord = true;
+              lineNumber += 1;
+
+              if (recordDelimiter != null) {
+                  // Check for how the line ends keying off what was provided in the label.
+                  // If the delimiter is "Carriage-Return Line-Feed" then the line should end with a carriage return and a line feed.
+                  if (recordDelimiter.equalsIgnoreCase("Carriage-Return Line-Feed") && !line.endsWith("\r\n")) {
+                    addTableProblem(ExceptionType.ERROR,
+                        ProblemType.MISSING_CRLF,
+                        "Record does not end in carriage-return line feed.",
+                        dataFile, tableIndex, reader.getCurrentRow());
+                    manuallyParseRecord = true;
+                  } else if (recordDelimiter.equalsIgnoreCase("Line-Feed") && !line.endsWith("\n")) {
```

Comment:
@qchaupds per my review comments, we need to improve this logic.

- [ ] update logic for handling CRLF line endings when LF is specified in the label

```
if recordDelimiter == Line-Feed:
   if !line.endsWith("\n"): THROW ERROR   // this is your current error thrown
   if line.endsWith("\r\n"): THROW ERROR   // this is a new error. line ends in CRLF not LF.
```